hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -74,9 +82,24 @@ public String getContext() {
         null : Arrays.copyOf(signature, signature.length);
   }
 
+  /**
+   * Whether the context is valid.
+   * The context should not contain '\t', '\n', '='.
+   * Because the context could be written to audit log.
+   */
   @InterfaceAudience.Private
   public boolean isContextValid() {
-    return context != null && !context.isEmpty();
+    if (context == null || context.isEmpty()) {
+      return false;
+    }
+
+    for (String str: ILLEGAL_CHARACTERS) {
+      if (context.contains(str)) {
+        return false;
+      }
+    }
+
+    return true;
```

Comment:
nit: could be replaced with `return ILLEGAL_CHARACTERS.stream().noneMatch(context::contains)`.